sdk/storage/azdatalake/filesystem/client.go

```diff
@@ -202,7 +203,15 @@ func (fs *Client) BlobURL() string {
 // The new directory.Client uses the same request policy pipeline as the Client.
 func (fs *Client) NewDirectoryClient(directoryPath string) *directory.Client {
 	directoryPath = strings.ReplaceAll(directoryPath, "\\", "/")
-	dirURL := runtime.JoinPaths(fs.generatedFSClientWithDFS().Endpoint(), directoryPath)
+	names := strings.Split(directoryPath, "/")
+	path := ""
+	for i, name := range names {
+		if i > 0 {
+			path += "/"
```

Comment:
In case of copy blob we fill the source path as well which goes as an query parameter, lets ensure encoding is correct there.

Reply:
verified the above